### ISSUE TITLE
schedule.scheduledJobs 

### ISSUE THREAD

**Issue by amnond** [node-schedule/node-schedule#313]
The scheduledJobs member of node-schedule is not mentioned in the documentation and also has a problem if two schedules with identical titles are added:

```

var schedule = require('node-schedule');

var rule = new schedule.RecurrenceRule();
rule.second = 10;

var j1 = schedule.scheduleJob("sched1", rule, function() { console.log("hi there 1"); });
var j2 = schedule.scheduleJob("sched1", rule, function() { console.log("hi there 2"); });

schedule.scheduledJobs['sched1'].cancel()  // j1 will continue to be invoked, but can't be accessed for cancelation

```

**Comment by santigimeno** [comment]
Both are known issues. PR's to improve the documentation are always welcome!

**Comment by amnond** [comment]
ok, thanks. Will try to help.

**Comment by connorzg** [comment]
@amnond @santigimeno is this solely an issue with .scheduledJobs? I am pushing all created jobs into a jobs array and canceling from there, though I am using the same variable name for scheduling (data is dynamic). Do you believe these jobs are still running?

**Comment by santigimeno** [comment]
@connorzg while you're storing `Job` objects and cancelling them via `Job.cancel()` there should not be any problem.

**Comment by connorzg** [comment]
Yep I've been able to run a lot of testing and ran into no issues. Thanks!

**Comment by rtodea** [comment]
Hi @santigimeno,

How should the scheduler behave in the scenario mentioned by @amnond?

I can imagine the following options:

1. throwing an exception
2. implicitly cancelling the previous job
 
The current behavior (i.e. keeping the first job active and having
access only to the second job) is counter-intuitive.

I had some issue in my project because of this.

Please advise.

Thanks!

**Comment by santigimeno** [comment]
@rtodea I agree. I think the throwing an exception makes more sense but both options are better than what we have now. What option do you prefer? Would you be willing to send a PR fixing the issue? From what I can tell both options would be `semver-major`

**Comment by rtodea** [comment]
@santigimeno, thanks for the reply!

I prefer option 2 (implicitly cancelling the previous job). 
I will come back with a PR for it.

**Comment by albertdatui** [comment]
@rtodea any updates for this?
If not I can help out with PR. (just want to avoid redundant effort).

**Comment by rtodea** [comment]
@albertdatui, I will provide an update these days (e.g. Friday).
Thanks for reminding me!

**Comment by jithu21** [comment]
Any update on this issue??

**Comment by santigimeno** [comment]
Not really, @rtodea was supposed to send a PR

**Comment by hayzem** [comment]
I am having a similiar problem. I prefer option 2 (implicitly cancelling the previous job). Is there any PR coming for it?

**Comment by deskoh** [comment]
Another issue is that job scheduled to run only once (i,e, [Date-based Scheduling](https://github.com/node-schedule/node-schedule#date-based-scheduling)) is never removed from `scheduledJobs`.
